server/lib/realtime/replication.ex

```diff
@@ -282,7 +282,7 @@ defmodule Realtime.Replication do
 
     %{
       state
-      | transaction: {lsn, %{txn | changes: Enum.reverse(truncated_relations ++ txn.changes)}}
+      | transaction: {lsn, %{txn | changes: Enum.reverse(truncated_relations) ++ txn.changes}}
```

Comment:
@soedirgo small optimization: 

```suggestion
      | transaction: {lsn, %{txn | changes: Enum.reverse(truncated_relations, txn.changes)}}
```

See: https://hexdocs.pm/elixir/Enum.html#reverse/2

Further optimization:

Since `process_message(%Truncate{} = msg, state)` is now doing `for` loop, `reverse`, and `concat`, we can combine all those operations like this:


```
  defp process_message(
         %Truncate{truncated_relations: truncated_relations},
         %State{transaction: {lsn, txn}} = state
       ) do
    truncated_relations_changes =
      generate_truncated_relations(state, truncated_relations, txn.changes)

    %{state | transaction: {lsn, %{txn | changes: truncated_relations_changes}}}
  end

  defp generate_truncated_relations(
         %State{relations: relations} = state,
         [truncated_relation | truncated_relations],
         acc
       ) do
    %{namespace: namespace, name: name} = Map.get(relations, truncated_relation)

    generate_truncated_relations(state, truncated_relations, [
      %TruncatedRelation{
        type: "TRUNCATE",
        schema: namespace,
        table: name
      }
      | acc
    ])
  end

  defp generate_truncated_relations(_state, [], acc) do
    acc
  end
```

Reply:
> Small optimization

Wow, TIL!

> Since process_message(%Truncate{} = msg, state) is doing for loop, reverse, and concat, we can combine all those operations like this

Hmm, the imperative code is much clearer to me. Truncate is not in the hot path, so clarity trumps here.